roles/openshift_master_upgrade/tasks/main.yml

```diff
@@ -0,0 +1,49 @@
+---
+# input variables:
+# - openshift_master_upgrade_pre_hook
+# - master_config_hook
+# - openshift.common.config_base
+# - openshift_master_upgrade_hook
+# - openshift.common.rolling_restart_mode
+# - openshift_master_upgrade_post_hook
+
+# Run the pre-upgrade hook if defined:
+- debug: msg="Running master pre-upgrade hook {{ openshift_master_upgrade_pre_hook }}"
+  when: openshift_master_upgrade_pre_hook is defined
+
+- include: "{{ openshift_master_upgrade_pre_hook }}"
+  when: openshift_master_upgrade_pre_hook is defined
+
+- include: rpm_upgrade.yml component=master
```

Comment:
Instead of doing in an include to update a single package here, it might make sense to either push this task file into a common parent role, so that it can be included from other upgrade roles as well.

The other thing that I'm wondering about is if we can consolidate this task with the similar tasks needed for updating docker containers (and eventually system containers). If the task file to be included is a variable, we can take advantage of dynamic includes and can avoid the performance hit of the skipped tasks for the deployment scenarios that do not apply. http://docs.ansible.com/ansible/playbooks_roles.html#dynamic-versus-static-includes